test/PositionToken.ts

```diff
@@ -1,15 +1,116 @@
-        positionToken = await ptContract.deploy();
+    describe("Roles and initialization", async function () {
+        before(async () => {
+            r = await buildContractTestContext();
+            ptContract = await ethers.getContractFactory("PositionToken");
+        });
+
+        it("Should be built properly by ContractTestContextt", async function () {
+            expect(r.positionToken).to.not.be.undefined;
+        });
+
+        it("Should not allow non admin to init", async function () {
+            const addr1PositionToken = await ptContract.deploy(r.addr1.address) as PositionToken;
+            const initPromise = addr1PositionToken.init(r.addr2.address);
+            await expect(initPromise).to.be.revertedWith("onlyAdmin: Caller is not admin");
+        });
+
+        it("Should not allow non executive to mint", async function () {
+            await r.positionToken.init(r.addr1.address);
+            const mintPromise = r.positionToken.safeMint(r.addr1.address);
+            await expect(mintPromise).to.be.revertedWith("onlyExecutive: Caller is not executive");
+        });
     });
 
-    it("Should exist", async function () {
-        expect(positionToken).to.not.be.undefined;
+    describe("After initialized with owner set to executive", async function () {
+        const firstTokenId = 0;
+        const secondTokenId = 1;
+        const thirdTokenId = 2;
+        let firstTokenOwner: SignerWithAddress;
+        let firstTokenOwnerAddress: string;
+        let secondTokenOwner: SignerWithAddress;
+        let secondTokenOwnerAddress: string;
+        let thirdTokenOwner: SignerWithAddress;
+        let thirdTokenOwnerAddress: string;
+
+        before(async function () {
+            r = await buildContractTestContext();
+            await r.positionToken.init(r.owner.address);
+            firstTokenOwner = r.addr1;
+            firstTokenOwnerAddress = r.addr1.address;
+            secondTokenOwner = r.addr2;
+            secondTokenOwnerAddress = r.addr2.address;
+            thirdTokenOwner = r.addr3;
+            thirdTokenOwnerAddress = r.addr3.address;
+        });
+
+        it("Should revert if the token id doesn't exist", async function () {
+            await expect(r.positionToken.ownerOf(0)).to.be.reverted;
+        });
+
+        it("Should be mintable from address designated executive", async function () {
+            await r.positionToken.safeMint(r.addr1.address);
+            expect(await r.positionToken.ownerOf(firstTokenId)).to.equal(firstTokenOwnerAddress);
+        });
+
+        it("Should increment the positionTokenId properly", async function () {
+            await r.positionToken.safeMint(r.addr2.address);
+            expect(await r.positionToken.ownerOf(secondTokenId)).to.equal(secondTokenOwnerAddress);
+        });
+
+        it("Should not be mintable from an address other than LeverageEngine", async function () {
+            expect(r.positionToken).to.not.be.undefined;
```

Comment:
not what the test says

Reply:
good catch

Reply:
6f0662117d6387314ffee6bf2b9d23301dbbb06a